### PR TITLE
[WINC-921]: add fast-forward workflow 

### PR DESCRIPTION
**Pull request by alinaryan**
This PR adds a windows fast-forward workflow and changes the config to use the new workflow.

### CODE REVIEW

File: ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    requests:
+      cpu: 100m
+      memory: 100Mi
+  credentials:
+  - namespace: test-credentials
+    name: acm-cicd-github
+    mount_path: /etc/acm-cicd-github
+  env:
+  - name: GITHUB_USER
+    default: "acm-cicd-prow-bot"
```
Comment (alinaryan) [inline ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-ref.yaml:18]: maybe use openshift-merge-robot?

**Comment by mtnbikenc** [comment]
No rehearsals have been run.  Not sure why the bot 'ack'ed this.
/pj-rehearse reject

**Comment by aravindhp** [comment]
@mtnbikenc we need to introduce a test that uses the workflow for rehearsals to work.

**Comment by mtnbikenc** [comment]
@aravindhp Are you saying that we need to convert this to a presubmit first to test it?
https://docs.ci.openshift.org/docs/how-tos/contributing-openshift-release/#marking-a-job-as-rehearsable

> Only the jobs with pj-rehearse.openshift.io/can-be-rehearsed: "true" label are rehearsed. Handcrafted jobs can opt to be rehearsable by including this label. All presubmits and periodics generated by make jobs have this label by default. **Generated postsubmits** will not contain it because generated postsubmits are used for promoting images.

**Comment by aravindhp** [comment]
@mtnbikenc no I am not suggesting we convert to a pre-submit. I did not release a [job](https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master-postsubmits.yaml) was already present. I guess we cannot do rehearsals on a post-submits. We can merge and test it out. Worst case I can delete the branch and recreate it.

**Review by mtnbikenc — CHANGES_REQUESTED** [review]
Thanks for working on this @alinaryan.  Just a few updates to the step ref for the secrets I created.

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

Nice work putting this all together

**Comment by mtnbikenc** [comment]
I tested the script locally and included `--dry-run` for the `git push` commands.
LGTM
[fast-forward.txt](https://github.com/openshift/release/files/10570924/fast-forward.txt)

File: ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    exit 1
+fi
+
+echo "INFO Cloning $DESTINATION_BRANCH"
+token=$(cat "$GITHUB_TOKEN_FILE")
+repo="github.com/$REPO_OWNER/$REPO_NAME"
+repo_url="https://${GITHUB_USER}:${token}@${repo}.git"
+
+if ! git clone -b "$DESTINATION_BRANCH" "$repo_url" ; then
+    echo "INFO $DESTINATION_BRANCH does not exist. Will create it"
```
Comment (aravindhp) [inline ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh:47]: Does this mean we no longer have to manually created `community-4.*` branches?

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @alinaryan.

File: ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh [hunk 3]
```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+wmco_dir=$(mktemp -d -t wmco-XXXXX)
+cd "$wmco_dir" || exit 1
+export HOME="$wmco_dir"
```
[the remark this answers is not in the document]
Comment (alinaryan) [reply ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh:5]: Good point, $HOME is not used again anywhere else. I ran it locally without this line and it was fine.

File: ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh (hunk 2 above)
Comment (alinaryan) [reply ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh:47]: Yes, is that ok? Are there steps missing that you usually follow when creating a community branch?

Comment (aravindhp) [reply ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh:47]: That should be ok. I guess we can find out by deleting the community-4.13 branch after this PR merges 😄

File: ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh [hunk 4]
```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
```
[the remark this answers is not in the document]
Comment (alinaryan) [reply ci-operator/step-registry/windows/ci/fastforward/windows-ci-fastforward-commands.sh:3]: Added

**Review by aravindhp — APPROVED** [review]
Nice work

**Pull request merged**